File: JetpacReloaded/Typescript version/HTML5Canvas/ts/jetman.ts
import { Color } from "../ts_lib/color.js";
import { Rectangle } from "../ts_lib/rectangle.js";
import { SpriteBatch, SpriteEffects, Texture2D } from "../ts_lib/spritebatch.js";
import { Vector2 } from "../ts_lib/vector2.js";
import { BaseObject } from "./baseobject.js";

export class Jetman extends BaseObject {
    private m_flip: SpriteEffects = SpriteEffects.None;
    private m_debugImage: Texture2D;

    public constructor(x: number, y: number, image: Texture2D, debugimage: Texture2D) {
        super();
        this.m_image = image;
        this.m_height = 68;// image.Height;
        this.m_width = 64;// image.Width / 5;
        this.m_rect = new Rectangle(this.m_frame * this.m_width, 0, this.m_width, this.m_height);
        this.m_screenLocation = new Vector2(x, y);
        this.m_debugImage = debugimage;
    }

    // public void Update(int x, int y, SpriteEffects flip)
    public Update(vec: Vector2, flip/*:SpriteEffects*/): void {
        //m_screenLocation.X = x;
        //m_screenLocation.Y = y;

        this.m_screenLocation = vec;

        //if (y <= 50) { m_screenLocation.Y = 50; }
        //if (y >= 550) { m_screenLocation.Y = 550; }
        //if (x <= 0) { m_screenLocation.X = 0; }
        //if (x >= 750) { m_screenLocation.X = 750; }

        this.m_rect = new Rectangle(this.m_frame * this.m_width, 0, this.m_width, this.m_height);
        this.m_flip = flip;
    }

    public Draw(spriteBatch: SpriteBatch): void {
        // // spriteBatch.Draw(this.m_debugImage, this.m_screenLocation, this.JetmanRect, Color.White, 0, Vector2.Zero, 1.0, this.m_flip, 0);
        // spriteBatch.Draw(this.m_image, this.m_screenLocation.X, this.m_screenLocation.Y);
        spriteBatch.Draw(this.m_image, this.m_screenLocation, this.m_rect, Color.White, 0, Vector2.Zero, 1.0, this.m_flip, 0);
    }

    public get JetmanRect(): Rectangle {
        return new Rectangle(this.m_screenLocation.X, this.m_screenLocation.Y, this.m_width, this.m_height);
    }

    public get JetmanPosition(): Vector2 { return this.m_screenLocation; }

    public set JetmanAnimFrame(value) { this.m_frame = value; }
}

File: JetpacReloaded/C# version/JetpacReloaded/Jetman.cs
using Microsoft.Xna.Framework;
using Microsoft.Xna.Framework.Graphics;

namespace JetPacReloaded
{
    public class Jetman : BaseObject
    {
        private SpriteEffects m_flip = SpriteEffects.None;
        private Texture2D m_debugImage;

        public Jetman(int x, int y, Texture2D image, Texture2D debugimage)
        {
            m_image = image;
            m_height = 68;// image.Height;
            m_width = 64;// image.Width / 5;
            m_rect = new Rectangle(m_frame * m_width, 0, m_width, m_height);
            m_screenLocation = new Vector2(x, y);
            m_debugImage = debugimage;
        }

        //        public void Update(int x, int y, SpriteEffects flip)
        public void Update(Vector2 vec, SpriteEffects flip)
        {
            //m_screenLocation.X = x;
            //m_screenLocation.Y = y;

            m_screenLocation = vec;

            //if (y <= 50) { m_screenLocation.Y = 50; }
            //if (y >= 550) { m_screenLocation.Y = 550; }
            //if (x <= 0) { m_screenLocation.X = 0; }
            //if (x >= 750) { m_screenLocation.X = 750; }

            m_rect = new Rectangle(m_frame * m_width, 0, m_width, m_height);
            m_flip = flip;
        }

        public new void Draw(SpriteBatch spriteBatch)
        {
            spriteBatch.Draw(m_debugImage, m_screenLocation, JetmanRect, Color.White, 0, Vector2.Zero, 1.0f, m_flip, 0);
            //spriteBatch.Draw(m_image, m_screenLocation, m_rect, Color.White, 0, Vector2.Zero, 1.0f, m_flip, 0);
        }

        public Rectangle JetmanRect
        {
            get { return new Rectangle((int)m_screenLocation.X, (int)m_screenLocation.Y, m_width, m_height); }
        }

        public Vector2 JetmanPosition { get { return m_screenLocation; } }

        public int JetmanAnimFrame { set { m_frame = value; } }




    }
}
